### DISCUSSION TITLE
执行abaqus脚本文件外代码

### DISCUSSION THREAD

**Discussion by Crook-cyber** [haiiliin/abqpy#5838] [Q&A]
将abaqus脚本写成一个包如下
from model_arm import creat_model
在另外一个.py文件导入包并运行无需打开abaqus的python脚本代码（通过abaqus cae -noGUI model_arm.py实现），除了能执行model_arm.py的代码以外，其他代码都不能执行
![image](https://github.com/user-attachments/assets/0a207a2b-6391-4cf6-b733-c4d5ec200215)
![image](https://github.com/user-attachments/assets/8f414e7f-e764-4031-8d43-bed4a11581af)

**Comment by haiiliin** [comment]
试试将 `model_arm` 模块的路径插入 `sys.path`:
```python
import os, sys

sys.path.insert(0, os.path.dirname(__file__))

from model_arm import creat_model

...
```

当 `model_arm.py` 和当前文件在同一文件夹时

**Reply by Crook-cyber** [reply]
不太行
